### PR TITLE
fix(cli): allow HTTPS registry URLs in native CLI binary

### LINKED ISSUE #8958: Native CLI rejects HTTPS Registry URLs because URL protocol handler is not enabled
## Description

The native Apicurio Registry CLI fails when configured with an HTTPS Registry API URL.

The failure occurs while validating the Registry URL, before the Vert.x HTTP client is created.

Observed error:

```text
Error: Could not create Registry client: Registry API URL
'https://apicurio.example.com/apis/registry/v3' is not well-formed:
Accessing a URL protocol that was not enabled.

The URL protocol https is supported but not enabled by default.
It must be enabled by adding the --enable-url-protocols=https option
to the native-image command.
```

## Environment

* Apicurio Registry server: `3.2.5-r1`
* CLI binary used for reproduction: upstream `3.2.4` Linux native CLI
* Registry URL: HTTPS OpenShift Route
* Authentication: anonymous
* Platform: Linux container running in Tekton on OpenShift

The `3.2.5` source appears to contain the same native-image configuration and URL-validation implementation as `3.2.4`.

## Steps to reproduce

Download and extract the native CLI, then configure a context using an HTTPS Registry URL:

```bash
acr config create-context test \
  --registry-url https://apicurio.example.com/apis/registry/v3
```

Alternatively, run any command that creates a Registry client using that context.

The CLI fails before making an HTTP request.

## Expected behavior

The native CLI should accept an HTTPS Registry API URL and connect using the Vert.x client.

## Actual behavior

The CLI rejects the HTTPS URL during client creation with a GraalVM URL protocol error.

## Suspected cause

The CLI native-image configuration currently contains:

```properties
quarkus.native.additional-build-args=\
  --initialize-at-run-time=io.vertx.ext.auth.impl.jose.JWT
```

It does not appear to enable the JDK HTTPS URL handler.

At the same time, Registry URL validation calls:

```java
var uri = new URI(url);
var ignored = uri.toURL();
```

This invokes the JDK `URL` protocol handler before the Vert.x adapter is selected.

In a native executable, `URI.toURL()` fails for `https:` unless that URL protocol handler was included at native-image build time.

Relevant files in the `3.2.5` tag:

```text
cli/src/main/resources/application.properties
java-sdk/common/src/main/java/io/apicurio/registry/client/common/RegistryClientRequestAdapterFactory.java
```

The latter catches the exception from `URI.toURL()` and reports the URL as “not well-formed,” even though the URL syntax itself is valid.

## Possible fixes

One option is to enable HTTPS URL handling in the native CLI build:

```properties
quarkus.native.enable-https-url-handler=true
```

or with the corresponding native-image argument:

```text
--enable-url-protocols=https
```

A potentially cleaner option would be to avoid calling `URI.toURL()` for syntax validation. The code could validate:

* that the value parses as a `URI`;
* that the scheme is `http` or `https`;
* that a host is present.

That would avoid depending on JDK URL protocol handlers when the actual HTTP implementation is Vert.x.

### PR DESCRIPTION
**Pull request by jsenko**
Closes #8958

## Summary

- Replace `URI.toURL()` validation with URI-only checks (scheme + host) in `RegistryClientRequestAdapterFactory` to avoid depending on JDK URL protocol handlers unavailable in GraalVM native images
- Enable `quarkus.native.enable-https-url-handler=true` in the CLI native build as defense-in-depth
- Add `RegistryUrlValidationTest` with 10 parameterized tests covering valid HTTPS/HTTP URLs, unsupported schemes, missing scheme, null/blank URLs, and malformed syntax

## Root Cause

The URL validation in `validateAndNormalizeRegistryUrl()` called `new URI(url).toURL()`. In a GraalVM native image, `URI.toURL()` fails for `https:` unless the HTTPS URL protocol handler was included at build time. The actual HTTP transport is Vert.x, so the JDK `URL` class was never needed — it was only used for validation.

## Test plan

- [x] Unit tests: `RegistryUrlValidationTest` — 10 tests covering valid and invalid URL patterns
- [x] Checkstyle passes on both modified modules
- [x] Manual verification: CLI accepts HTTPS URLs and connects (SSL handshake succeeds with trusted certs, fails correctly with self-signed — no more "URL protocol https is not enabled" error)
- [ ] Native binary build verification (CI)

### CODE REVIEW

File: java-sdk/common/src/main/java/io/apicurio/registry/client/common/RegistryClientRequestAdapterFactory.java [hunk 1]
```diff
@@ -305,9 +306,16 @@ private static String validateAndNormalizeRegistryUrl(RegistryClientOptions opti
...
-        } catch (Exception ex) {
+            var uri = new URI(url);
+            if (uri.getScheme() == null || uri.getHost() == null) {
+                throw new IllegalArgumentException(
+                        "Registry API URL '" + url + "' is not well-formed: scheme and host are required.");
+            }
+            if (!"http".equals(uri.getScheme()) && !"https".equals(uri.getScheme())) {
+                throw new IllegalArgumentException(
+                        "Registry API URL '" + url + "' is not well-formed: unsupported scheme '" + uri.getScheme() + "'.");
+            }
```
[the remark this answers is not in the document]
Comment (jsenko) [reply java-sdk/common/src/main/java/io/apicurio/registry/client/common/RegistryClientRequestAdapterFactory.java:318]: Will fix. `java.net.URI` does not normalize scheme case, so `HTTP://` would have been rejected. Added `toLowerCase(Locale.ROOT)` before comparison. Fixed in d844d00.

File: java-sdk/common/src/test/java/io/apicurio/registry/client/common/RegistryUrlValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    })
+    void testValidUrlsAreAccepted(String url) {
+        var options = RegistryClientOptions.create().rawRegistryUrl(url);
+        try {
+            RegistryClientRequestAdapterFactory.createRequestAdapter(options, Version.V3);
+        } catch (IllegalArgumentException ex) {
+            throw new AssertionError("URL '" + url + "' should be accepted but was rejected: " + ex.getMessage(), ex);
+        } catch (Exception ignored) {
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jsenko) [reply java-sdk/common/src/test/java/io/apicurio/registry/client/common/RegistryUrlValidationTest.java:27]: Will fix. Narrowed the catch to `IllegalStateException` only (thrown when adapter module is missing from classpath). Any other unexpected exception will now correctly fail the test. Fixed in d844d00.

**Comment by carlesarnal** [comment]
Re-review after d844d00: Both previous items are fixed. Case-insensitive scheme comparison uses toLowerCase(Locale.ROOT) correctly. Test catch clause narrowed to IllegalStateException only. Two minor non-blocking suggestions: (1) consider importing java.util.Locale instead of using the FQN inline at line 314, and (2) add a test case with uppercase scheme like HTTPS://example.com to exercise the toLowerCase path. Overall looks good.
